dgsh-tee deadlock

tammam1998 opened the pull request:
This is an attempt to fix issue #116  by moving the termination code to be after reached_eof is set.

dspinellis commented:
Thank you for debugging and submitting this! Can you please ensure that no unneeded parts are modified? It's difficult to see what you fixed in the [current diff](https://github.com/dspinellis/dgsh/pull/117/commits/c39d25fdc6f31a42c2c26cf3b00cbe61ec702f4b). Feel free to rebase and force-push a single commit.

tammam1998 committed:
Attempt to fix deadlock on select in dgsh-tee

tammam1998 commented:
Hi @dspinellis I squashed the commits. I realized that my old fix didn't pass the test cases for some reason so I changed it. Changes should be clearer now. I'm not sure if there is a better way to go about fixing this issue but my fix is basically just making sure we set at least one fd before calling select.

dspinellis commented:
Good catch; thank you!
